core/vm/memory.go

```diff
@@ -85,8 +85,12 @@ func (m *Memory) Get(offset, size int64) (cpy []byte) {
 	return
 }
 
+// ReadOnlyMem represents a slice of memory which has not been copied,
+// but is a pointer directly to memory, and should not be modified
+type ReadOnlyMem []byte
+
 // GetPtr returns the offset + size
-func (m *Memory) GetPtr(offset, size int64) []byte {
+func (m *Memory) GetPtr(offset, size int64) ReadOnlyMem {
```

Comment:
Do we still use the `Get` version? If not, perhaps lets get rid of it to avoid reintroducing the copies later?

Reply:
Yes, 

- Log operations, 
- Create2 and Create, 
- Mload

Reply:
.... that being said.. it's useless to have `Mload` use `Get` as opposed to `GetPtr`, since all it does is send it into `SetBytes` on a big.Int, which copies it internally...